src/main/features/ccj/routes/check-and-send.ts

```diff
@@ -18,28 +22,69 @@ function prepareUrls (externalId: string): object {
 }
 
 function renderView (form: Form<Declaration>, req: express.Request, res: express.Response): void {
+  const user: User = res.locals.user
   res.render(Paths.checkAndSendPage.associatedView, {
     form: form,
-    details: res.locals.user.ccjDraft,
-    amountToBePaid: res.locals.user.claim.totalAmount - (res.locals.user.ccjDraft.paidAmount.amount || 0),
+    details: user.ccjDraft,
+    amountToBePaid: user.claim.totalAmount - (user.ccjDraft.paidAmount.amount || 0),
+    partyAsCompanyOrOrganisation: isPartyCompanyOrOrganisation(user),
     ...prepareUrls(req.params.externalId)
   })
 }
 
+function isPartyCompanyOrOrganisation (user: User): boolean {
+  const claim: Claim = user.claim
+  const type: string = claim.claimData.claimant.type
+  return type === PartyType.COMPANY.value || type === PartyType.ORGANISATION.value
+}
+
+function deserializerFunction (value: any): any {
+  switch (value.type) {
+    case SignatureType.BASIC:
+      return Declaration.fromObject(value)
+    case SignatureType.QUALIFIED:
+      return QualifiedDeclaration.fromObject(value)
+    default:
+      throw new Error(`Unknown declaration type: ${value.type}`)
+  }
+}
+
+function getStatementOfTruthClassFor (user: User): any {
+  if (isPartyCompanyOrOrganisation(user)) {
+    return QualifiedDeclaration
+  } else {
+    return Declaration
+  }
+}
+
+function signatureTypeFor (user: User): string {
+  if (isPartyCompanyOrOrganisation(user)) {
+    return SignatureType.QUALIFIED
+  } else {
+    return SignatureType.BASIC
+  }
+}
+
 export default express.Router()
-  .get(Paths.checkAndSendPage.uri, (req: express.Request, res: express.Response, next: express.NextFunction) => {
-    renderView(Form.empty<Declaration>(), req, res)
+  .get(Paths.checkAndSendPage.uri, (req: express.Request, res: express.Response) => {
+    const StatementOfTruthClass = getStatementOfTruthClassFor(res.locals.user)
+    renderView(new Form(new StatementOfTruthClass()), req, res)
   })
   .post(
     Paths.checkAndSendPage.uri,
-    FormValidator.requestHandler(Declaration, Declaration.fromObject),
+    FormValidator.requestHandler(undefined, deserializerFunction),
     ErrorHandling.apply(async (req: express.Request, res: express.Response, next: express.NextFunction) => {
-      const form: Form<Declaration> = req.body
+      const form: Form<any> = req.body
       const user: User = res.locals.user
 
       if (form.hasErrors()) {
         renderView(form, req, res)
       } else {
+        if (signatureTypeFor(user) === SignatureType.QUALIFIED) {
```

Comment:
It makes more sense for me to do:
```if (form.model.type === SignatureType.QUALIFIED) {```

Reply:
that's submitted in the form though, a malicious user could change it?

Reply:
done